iroh-net/src/dns.rs

```diff
@@ -14,23 +17,83 @@ fn get_resolver() -> Result<TokioAsyncResolver> {
     let (config, mut options) =
         trust_dns_resolver::system_conf::read_system_conf().unwrap_or_default();
     // lookup IPv4 and IPv6 in parallel
-    options.ip_strategy = trust_dns_resolver::config::LookupIpStrategy::Ipv4AndIpv6;
+    options.ip_strategy = trust_dns_resolver::config::LookupIpStrategy::Ipv4thenIpv6;
 
     let resolver = AsyncResolver::tokio(config, options);
     Ok(resolver)
 }
 
+/// Resolve IPv4 and IPv6 in parallel
+pub(crate) async fn lookup_ipv4_ipv6<N: IntoName + TryParseIp + Clone>(
+    host: N,
+    timeout: Duration,
+) -> Result<Vec<IpAddr>> {
+    let ipv4 = DNS_RESOLVER.ipv4_lookup(host.clone());
+    let ipv6 = DNS_RESOLVER.ipv6_lookup(host);
+    let ipv4 = tokio::time::timeout(timeout, ipv4);
+    let ipv6 = tokio::time::timeout(timeout, ipv6);
+
+    let res = futures::future::join(ipv4, ipv6).await;
+    match res {
+        (Ok(Ok(ipv4)), Ok(Ok(ipv6))) => {
+            let res = ipv4
+                .into_iter()
+                .map(|ip| IpAddr::V4(ip.0))
+                .chain(ipv6.into_iter().map(|ip| IpAddr::V6(ip.0)))
+                .collect();
+            Ok(res)
+        }
+        (Ok(Ok(ipv4)), Err(_err)) => {
```

Comment:
```suggestion
        (Ok(Ok(ipv4)), Err(_timeout)) => {
```
is perhaps a little bit more readable?
(in all places were a timeout err is matched)

Reply:
done